strax/plugin.py

```diff
@@ -409,8 +417,10 @@ def cleanup(self,
         # Check the input buffer is empty
         for d, buffer in self.input_buffer.items():
             if buffer is not None and len(buffer):
-                raise RuntimeError(
-                    f"Plugin {d} terminated with leftover {d}: {buffer}")
+                # This can happen especially in time range selections
+                if int(self.save_when) != strax.SaveWhen.NEVER:
```

Comment:
Yes, nice, I think this is the key fix. (You could move the if condition up above the for loop for clarity, since the loop doesn't need to run at all for never-save plugins)